CONTRIBUTING.md

```diff
@@ -15,12 +15,63 @@ us first through the issue tracker with your idea so that we can help out and
 frustration later on.
 
-### Code reviews
+## Code reviews
+
 All submissions, including submissions by project members, require review. We
 use Github pull requests for this purpose.
 
-### The small print
+The submitted code should adhere to the following:
+
+### Do's
+
+* For every pull request, make sure all commits are consolidated into one commit before
+  sending the pull request.
+* Before submitting any code changes, please ensure that the [Unit Tests](https://github.com/google/EarlGrey/tree/master/Tests/UnitTests)
+  and [Functional Tests](https://github.com/google/EarlGrey/tree/master/Tests/FunctionalTests)
+  projects build and all test pass. If you believe that the test failures are due to bugs in the
+  framework then please [file an issue](https://github.com/google/EarlGrey/issues).
+* If a pull request looks good then we'll wait for Travis to run all the Unit and
+  Functional tests on the changes. We also trigger an internal build to check for
+  further build breakages. In case any of the tests are broken, we will let you know.
+* If your pull request does not touch any code, then you could skip the Travis Continuous
+  Integration check by adding [ci skip](https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build)
+  to your git commit message. This helps prevent over-encumbering the CI.
+* Please use [Github's Markdown Formatting](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/)
+  for formatting your comments and issues.
+* Refer to the [Google Objective-C
+  style guide](https://google.github.io/styleguide/objcguide.xml) for our code conventions.
+
+### Don'ts
+
+* Consider not adding LGTM's or +1's unless you're specifically asked to do so, since they
+  add noise to the conversation.
+* Refrain from providing small and granular pull requests for changes that could easily be
+  consolidated into one.
+* A vague description for a pull request adds to the review time because we now have to analyze
+  what the code does. A more descriptive description will help the reviewers understand what
+  they're reviewing.
+
+## Github Issues
+
+### Do's
+
+* Ensure you've searched extensively through the documentation before filing a Github Issue. As
+  a developer, please provide us issue descriptions that you yourself would like to read.
+* Please be as verbose in your comments and descriptions as possible. For each issue, add in the
+  type of project, language used, Xcode version, error trace etc. While we love the fact that
+  you're helping us improve the project, it would be nice for us not to have to ask for debugging
+  clarifications for every issue created.
+* For in-depth discussions please use the [EarlGrey Google Group](https://groups.google.com/forum/#!forum/earlgrey-discuss)
+  instead of starting a Github Issue.
+* For new feature requests, show us how the impact of the new feature would justify any maintenance
```

Comment:
"...For new feature requests..."
is it necessary to include it and does it add much value?